src/commands/bso/fishingcontest.ts

```diff
@@ -0,0 +1,216 @@
+import { BotCommand } from '../../lib/structures/BotCommand';
+import { FishingContestOptions } from '../../lib/types/minions';
+import { formatDuration, stringMatches, updateBankSetting } from '../../lib/util';
+import addSubTaskToActivityTask from '../../lib/util/addSubTaskToActivityTask';
+import { calculateFishingContestXP } from '../../tasks/minions/bso/fishingContestActivity';
+
+export default class extends BotCommand {
+	public constructor(store: CommandStore, file: string[], directory: string) {
+		super(store, file, directory, {
+			oneAtTime: true,
+			altProtection: true,
+			categoryFlags: ['minion', 'minigame'],
+			description: 'Sends your minion to do the Fishing Contest minigame.',
+			examples: ['=fishingcontest [fish]'],
+			subcommands: true,
+			usage: '[fish] [input:...string]',
+			usageDelim: ' '
+		});
+	}
+
+	@requiresMinion
+	async run(msg: KlasaMessage) {
+		const currentFishType = getCurrentFishType();
+		const userDetails = await getUsersFishingContestDetails(msg.author);
+		const validLocs = getValidLocationsForFishType(currentFishType);
+		const minigameScore = await getMinigameScore(msg.author.id, 'fishing_contest');
+		return msg.channel.send(
+			`**Todays Catch:** A fish from a ${currentFishType.temperature} ${currentFishType.water} (${validLocs
+				.map(i => i.name)
+				.join(', ')})
+**Todays Catches:** ${userDetails.catchesFromToday
+				.sort((a, b) => b.length_cm - a.length_cm)
+				.map(i => `${i.name}(${i.length_cm / 100}m)`)
+				.join(', ')}
+**Total Daily Contests:** ${minigameScore}
+**All-time catches:** ${userDetails.catchesAllTime}
+**Total Unique Catches:** ${userDetails.totalUniqueCatches}
+**Total Length of Fish:** ${userDetails.totalLength / 100}m`
+		);
+	}
+
+	@minionNotBusy
+	@requiresMinion
+	async fish(msg: KlasaMessage, [input = '']: [string | undefined]) {
+		const fishingLocation = fishingLocations.find(i => stringMatches(i.name, input));
+		if (!fishingLocation) {
+			return msg.channel.send(
+				`That's not a valid location to fish at, you can fish at these locations: ${fishingLocations
+					.map(i => `${i.name}(${i.temperature} ${i.water})`)
+					.join(', ')}.`
+			);
+		}
+		const currentFishType = getCurrentFishType();
+		const validLocs = getValidLocationsForFishType(currentFishType);
+		if (!validLocs.includes(fishingLocation)) {
+			return msg.channel.send(
+				`This Fishing Location isn't valid for todays catch! These ones are: ${validLocs
+					.map(i => i.name)
+					.join(', ')}`
+			);
+		}
+
+		if (!['Contest rod', "Beginner's tackle box"].every(i => msg.author.hasItemEquippedOrInBank(i))) {
+			return msg.channel.send(
+				'You need to buy a Contest rod and a tackle box to compete in the Fishing contest.'
+			);
+		}
+
+		if (msg.flagArgs.reset) {
+			await prisma.fishingContestCatch.deleteMany({
+				where: {
+					user_id: BigInt(msg.author.id)
+				}
+			});
+			await msg.author.settings.reset(UserSettings.CollectionLogBank);
```

Comment:
I don't think you intend for people to delete their entire CL with this command...